services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/pathverification/PathVerificationService.java

```diff
@@ -469,10 +510,28 @@ static long noviflowTimestamp(byte[] timestamp) {
         return seconds * TEN_TO_NINE + nanoseconds;
     }
 
+    /**
+     * Calculate latency of the ISL.
+     * @param switchId Switch ID of endpoint which received packet
+     * @param port port of endpoint which received packet
+     * @param switchT0 sent packet timestamp in nanoseconds
+     * @param switchT1 receive packet timestamp in nanoseconds
+     * @return long
+     */
+    @VisibleForTesting
+    static long calculateRoundTripLatency(SwitchId switchId, int port, long switchT0, long switchT1) {
+        if (switchT0 <= 0 || switchT1 <= 0) {
+            logger.warn("Invalid round trip latency timestamps on endpoint {}_{}. t0 = '{}', t1 = '{}'.",
```

Comment:
Do we need to handle this in a special case on the storm side. Need to figure out

Reply:
will handle this case on storm side